stores/sql.go

```diff
@@ -402,3 +393,43 @@ func (s *SQLStore) retryTransaction(fc func(tx *gorm.DB) error, opts ...*sql.TxO
 	}
 	return fmt.Errorf("retryTransaction failed: %w", err)
 }
+
+func initConsensusInfo(db *gorm.DB) (dbConsensusInfo, modules.ConsensusChangeID, error) {
+	var ci dbConsensusInfo
+	if err := db.
+		Where(&dbConsensusInfo{Model: Model{ID: consensusInfoID}}).
+		Attrs(dbConsensusInfo{
+			Model: Model{ID: consensusInfoID},
+			CCID:  modules.ConsensusChangeBeginning[:],
+		}).
+		FirstOrCreate(&ci).
+		Error; err != nil {
+		return dbConsensusInfo{}, modules.ConsensusChangeID{}, err
+	}
+	var ccid modules.ConsensusChangeID
+	copy(ccid[:], ci.CCID)
+	return ci, ccid, nil
+}
+
+func (s *SQLStore) ResetConsensusSubscription() error {
+	// drop tables
+	if err := s.db.Migrator().DropTable(&dbConsensusInfo{}, &dbSiacoinElement{}, &dbTransaction{}); err != nil {
+		return err
+	}
+	// recreate the tables.
+	err := s.db.Migrator().AutoMigrate(&dbConsensusInfo{}, &dbSiacoinElement{}, &dbTransaction{})
+	if err != nil {
+		return err
+	}
+	// initialise the consenus_info table.
+	ci, _, err := initConsensusInfo(s.db)
+	if err != nil {
+		return err
+	}
+	// reset in-memory state.
+	s.chainIndex = types.ChainIndex{
```

Comment:
we should probably move the `chainIndex` to be covered by the mutex and acquire it here before resetting it? I know it won't cause races but I think the `Reset` is done in a goroutine so it's best to cover it with a mutex.